CMakeLists.txt

```diff
@@ -156,17 +169,63 @@ if(NGEN_WITH_NETCDF)
     add_compile_definitions(NETCDF_ACTIVE)
 endif()
 
+if(NGEN_WITH_EXTERN_CXX_MODELS)
+    # This is set because SLOTH will link to gtest by default,
+    # which causes a configure error.
+    set(PACKAGE_TESTS OFF CACHE BOOL "SLoTH testing")
+    add_external_subdirectory(
+        SOURCE "${NGEN_EXT_DIR}/sloth"
+        OUTPUT "${NGEN_EXT_DIR}/sloth/cmake_build"
+        GIT_UPDATE "${NGEN_EXT_DIR}/sloth"
+    )
+
+    set_target_properties(slothmodel PROPERTIES
+        CXX_VISIBILITY_PRESET default)
+endif()
+
 # -----------------------------------------------------------------------------
 # Handle several steps for BMI C library logic and dependencies, at top level, if functionality is turned on
 if(NGEN_WITH_BMI_C)
     add_compile_definitions(NGEN_BMI_C_LIB_ACTIVE)
+
+    if(NGEN_WITH_EXTERN_C_MODELS)
+        add_external_subdirectory(
+            SOURCE "${NGEN_EXT_DIR}/topmodel"
+            OUTPUT "${NGEN_EXT_DIR}/topmodel/cmake_build"
+            GIT_UPDATE "${NGEN_EXT_DIR}/topmodel/topmodel"
```

Comment:
Would it be worthwhile to make each model/module a configuration option?  e.g. `NGEN_WITH_EXTERN_TOPMODEL`?

Reply:
Probably -- should these default to ON? (dependent on if ngen is built to support the model's underlying language)

Reply:
If each one is going to be an option, can we have a NGEN_WITH_EXTERN_ALL for the CIROH NGIAB?

Reply:
@ZacharyWills Yeah, I can make this option available. The semantics of it are a bit tricky, since models are dependent on other options (i.e. `NGEN_WITH_CFE` requires `NGEN_WITH_BMI_C`), so just need to make sure when `NGEN_WITH_EXTERN_ALL=ON`, then only the correct options are enabled:

i.e.
`NGEN_WITH_BMI_C=ON, NGEN_WITH_BMI_FORTRAN=OFF, NGEN_WITH_EXTERN_ALL=ON` implies `NGEN_WITH_CFE=ON, ..., NGEN_WITH_NOAH_OWP_MODULAR=OFF`.

Reply:
@ZacharyWills Added suggestion in d265dea.